orchestrator/forms/__init__.py

```diff
@@ -9,153 +9,3 @@
-    validator_name: str
-    errors: List[PydanticErrorDict]
-
-    def __init__(self, validator_name: str, errors: List[Dict[str, Any]]):
-        super().__init__(validator_name, errors)
-        self.validator_name = validator_name
-        self.errors = cast(List[PydanticErrorDict], errors)
-
-    def __str__(self) -> str:
-        no_errors = len(self.errors)
-        return (
-            f'{no_errors} validation error{"" if no_errors == 1 else "s"} for {self.validator_name}\n'
-            f"{display_errors(cast(List[Dict[str, Any]], self.errors))}"  # type: ignore
-        )
-
-
-def generate_form(
-    form_generator: Optional[StateInputFormGenerator], state: State, user_inputs: List[State]
-) -> Optional[InputForm]:
-    """Generate form using form generator as defined by a workflow."""
-    try:
-        # Generate form is basically post_process
-        post_process(form_generator, state, user_inputs)
-    except FormNotCompleteError as e:
-        # Form is not finished and raises the next form, this is expected
-        return e.form
-
-    # Form is finished and thus there is no new form
-    return None
-
-
-def post_process(form_generator: Optional[StateInputFormGenerator], state: State, user_inputs: List[State]) -> State:
-    """Post process user_input based on form definition from workflow."""
-
-    # there is no form_generator so we return no validated data
-    if not form_generator:
-        return {}
-
-    current_state = deepcopy(state)
-
-    logger.debug("Post process form", state=state, user_inputs=user_inputs)
-
-    # Generate generator
-    generator = form_generator(current_state)
-
-    try:
-        # Generate first form (we need to send None here, since the arguments are already given when we generated the generator)
-        generated_form: InputForm = generator.send(None)
-
-        # Loop through user inputs and for each input validate and update current state and validation results
-        for user_input in user_inputs:
-            # Validate
-            try:
-                form_validated_data = generated_form(**user_input)
-            except ValidationError as e:
-                raise FormValidationError(e.model.__name__, e.errors()) from e  # type: ignore
-
-            # Update state with validated_data
-            current_state.update(form_validated_data.dict())
-
-            # Make next form
-            generated_form = generator.send(form_validated_data)
-
-        else:
-            # Form is not completely filled raise next form
-            raise FormNotCompleteError(generated_form.schema())
-    except StopIteration as e:
-        # Form is completely filled so we can return the last of the data and
-        return e.value
-
-
-class DisplayOnlyFieldType:
-    @classmethod
-    def __get_validators__(cls) -> Generator:
-        yield cls.nothing
-
-    def nothing(cls, v: Any, field: ModelField) -> Any:
-        return field.default
-
-
-class FormPage(BaseModel):
```

Comment:
I've tried running mypy on our surf orchestrator when using this branch of the orchestrator-core, and this was the result:

![image](https://github.com/workfloworchestrator/orchestrator-core/assets/1482100/a8112d58-a0a3-4482-af82-488ded078d5a)

I didn't really consider this upfront. :) 

We can rewrite all of these, but this effort will also be required from other users of the orchestrator. We'd at least need to bump the package version to `1.3.x` to signal these breaking changes.

For the sake of backwards compatibility and making it easier to migrate, we could consider to import the pydantic_forms types/functions in orchestrator so that the import paths don't change.


For example:
```
# orchestrator/forms/__init__.py

from pydantic_forms.core import FormPage  # noqa: F401
```
(F401 prevents linters from removing the unused import)

This would allow users of the orchestrator-core package to keep doing `from orchestrator.forms import FormPage`, and postpone having to change the imports. (perhaps we can create our own code-migration tool for it someday, similar to https://github.com/asottile/pyupgrade and https://github.com/pydantic/bump-pydantic)

What do you think?

Reply:
Apart of having backwards-compatible imports (or not), we'll at least need a summary in the release notes of which symbols have been removed, and tell users where they can find them in pydantic_forms.

From what I can see in our surf orchestrator mypy results, these are the validators/types that we depend on

```
orchestrator.forms.FormPage
orchestrator.forms.ReadOnlyField
orchestrator.forms.validators.Accept
orchestrator.forms.validators.Choice
orchestrator.forms.validators.ContactPersonList
orchestrator.forms.validators.DisplaySubscription
orchestrator.forms.validators.Divider
orchestrator.forms.validators.Label
orchestrator.forms.validators.ListOfOne
orchestrator.forms.validators.ListOfTwo
orchestrator.forms.validators.LongText
orchestrator.forms.validators.MigrationSummary
orchestrator.forms.validators.OrganisationId
orchestrator.forms.validators.UniqueConstrainedList
orchestrator.forms.validators.choice_list
orchestrator.forms.validators.contact_person_list
orchestrator.forms.validators.migration_summary
orchestrator.types.InputForm
orchestrator.types.FormGenerator
orchestrator.types.SummaryData
```